auth/auth.go

```diff
@@ -81,6 +87,17 @@ func (a ACL) VerifyPeerCertificateServer(rawCerts [][]byte, verifiedChains [][]*
 		return nil
 	}
 
+	// Check UID against --allow-uid flag(s).
+	if len(a.AllowedUIDs) > 0 {
+		for _, typeVal := range cert.Subject.Names {
+			if typeVal.Type.String() == uidOID {
+				if contains(a.AllowedUIDs, typeVal.Value.(string)) {
```

Comment:
Type assertion here could panic, add a check to make sure the typecast was successful. Also perhaps we could extract the logic to read UIDs from certificates into a separate function, like `uidNames(cert *x509.Certificate) []string`?